babel/board.js

```diff
@@ -46,12 +47,48 @@ export default class Board {
         return _.uniq(_.flatten(this.orbs)).sort();
     }
 
-    /**
-     * @returns {Object} - A match event. Includes the
-     *
-     */
+    findMatches() {
+        let matches = [],
+            b = _.cloneDeep(this.orbs);
+        
+        _.each(_.range(this.height - 2), x => {
+            _.each(_.range(this.width - 2), y => {
+                if (b[x][y] == b[x][y + 1] && b[x][y] == b[x][y + 2]) {
+                    matches.push([[x, y], [x, y + 1], [x, y + 2]])
+                } else if (b[x][y] == b[x + 1][y] && b[x][y] == b[x + 2][y]) {
+                    matches.push([[x, y], [x + 1, y], [x + 2, y]])
+                };
+            });
+        });
+        return matches;
+    }
+    
+    combineMatches() {
+        let combinedMatches = [],
+            unused = _.clone(this.findMatches()), 
+            couldMatch, before, currentMatch;
+        
+        while (unused[0] != null) {
```

Comment:
I think this is simpler read as

```js
while (unused.shift()) {
    // ...
}
```

that'll break from the while loop once `shift()` returns null.